### DISCUSSION TITLE
[Application] Slate

### DISCUSSION THREAD

**Discussion by jennijuju** [filecoin-project/community#60] [Project Showcase]
Learn about https://slate.host, built by @slate-engineering, [here](https://github.com/filecoin-project/slate).

**Comment by jimmylee** [comment]
## Previews
![Et-LX1aXUAktNaR](https://user-images.githubusercontent.com/310223/108011611-48bfdb80-6fbc-11eb-9922-41e0818d1bb1.jpg)

![Et5UUjmXEAoCYxH](https://user-images.githubusercontent.com/310223/108011600-42c9fa80-6fbc-11eb-89e2-1d2d0d88d57f.jpg)

## Updates
For the week ending on Friday, Feb 12th

* We had a conversation with @andyschwab (Protocol Labs), @whyrusleeping (Protocol Labs) and @Schwartz10 (Infinite Scroll) about using their DiD solution through Daemon Land. It isn't conclusion yet but there is a video upon request if you want to learn more details about how it could work. Integrating Daemon Land will solve what @autonome (Protocol Labs) requested back in October 28th, 2020 https://github.com/filecoin-project/slate/issues/390
* We spoke with https://twitter.com/dankvr from https://github.com/webaverse/app about integrating Slate, we look forward to supporting their cases better.
* I gave a brief masterclass on NFTs to the Slate team, its based on this GitHub issue. https://github.com/filecoin-project/slate/issues/523
* @jasonleyser and @tarafanlin launched the Chrome Extension V2
* @Aminejvm added drag and drop selection to make experience feel like your file system.
* @akuokojnr integrated an online indicator to test our real time.
* @martinalong has made significant progress on the iOS application.
* @harisbutt is currently re-working our design system and working with @cwaring (Protocol Labs) on an editor. @Aminejvm has markdown posts showing in the UI.

## Social
Notable social posts for the week.

* https://twitter.com/_akuokojnr/status/1359617459377102852
* https://twitter.com/AmineJv/status/1359963053442367499
* https://twitter.com/jason_leyser/status/1360365101623070721
* https://twitter.com/agata_krych/status/1359617852165259267

**Comment by martinalong** [comment]
![IMG_8468](https://user-images.githubusercontent.com/33686587/108571671-89259f00-72c5-11eb-8a08-e22336f0f183.JPG)

![image](https://user-images.githubusercontent.com/33686587/108571832-df92dd80-72c5-11eb-95cd-02239b92984f.png)

## Updates
For the week ending on Friday, Feb 19th
- @akuokojnr added logic to open pdf files in a new tab on mobile devices, to solve the pdf mobile rendering issue. And fixed upload to slate bug that wasn't adding it to the selected slate
- @aminejvm added markdown renderer in slate to render text files
- @tarafanlin worked on the chrome extension: product tour, UI for setting’s page and part of file uploader
- @toastt added filter buttons in files folder
- @harisbutt worked on filter buttons and presentation deck 
- @martinalong added activity and explore to the mobile app. Got uploads to textile working

## Social
- https://twitter.com/aminejv/status/1362900724158464007?s=21

**Comment by martinalong** [comment]
![image](https://user-images.githubusercontent.com/33686587/109432548-90813280-79c0-11eb-9af4-03ad9ea6e618.png)

![image](https://user-images.githubusercontent.com/33686587/109432555-9aa33100-79c0-11eb-9fbd-8e4d1828e5de.png)

![image](https://user-images.githubusercontent.com/33686587/109432565-a7278980-79c0-11eb-9318-e25ac276ef7e.png)

## Updates
For the week ending on Friday, Feb 26th

- @martinalong worked on loading activity on scroll to bottom or pull down to refresh, and multi upload and document upload for the iOS app
- @akuokojnr added logic to show the slate object url in the activity tab, 3D card component for the slate design system, wrote and researched a post on NFT integration in slate 
- @aminejvm worked on metamask integration with slate, explored NFT support options for slate
- @tarafanlin worked on chrome extension local storage setup, design changes for extension upload history
- @cwaring worked on linting config and dependency management, UI bug fixes, link object support [#575](https://github.com/filecoin-project/slate/pull/575)
- @toastts worked on reading render and imgproxy docs, looking for good solution to deploy imgproxy on render

## Social
- https://twitter.com/toastts_/status/1364028980878172161?s=20
- https://twitter.com/astrangel00p/status/1363241915743834113?s=20

**Comment by martinalong** [comment]
![image](https://user-images.githubusercontent.com/33686587/110192890-b43fe080-7de5-11eb-8308-f5f8785a2865.png)

![image](https://user-images.githubusercontent.com/33686587/110192927-dd607100-7de5-11eb-9ac6-d3f512181feb.png)

![image](https://user-images.githubusercontent.com/33686587/110192959-07199800-7de6-11eb-88d1-098f03bb1151.png)


### Updates
For the week ending on Friday, Mar 5th

- @tarafanlin worked on chrome extension UI for uploads and popup, and finalizing and implementing local storage
- @aminejvm worked on multi-file downloading for selected files and whole slates
- @jasonleyser shipped multi image uploads for chrome extension (where you can leave the page while its uploading) and API verification in welcome page
- @martinalong worked on search in iOS and implementing new team operating procedures
- @harisbutt worked on exploring future routes for slate and new team operating procedures

**Comment by martinalong** [comment]
![image](https://user-images.githubusercontent.com/33686587/111013706-47809500-8355-11eb-836b-ab77c55dfd57.png)

![image](https://user-images.githubusercontent.com/33686587/111013731-6717bd80-8355-11eb-8a3c-57edc34d5ced.png)

### Updates

For the week ending on Friday, Mar 12th
- @akuokojnr cleaned up and corrected imports in the slate design system
- @aminejvm built analytics dashboard using metabase
- @jasonleyser explored partnerships that can help us add content from partners like museums, blogs, etc directly into slate. Also worked on extension testing and bug fixes
- @cwaring extended link objects to support NFT support
- @tarafanlin worked on slate tagging design, sign-in and onboarding design, and nft.storage website design
- @toastts worked on API page redeisgn and programatically adding content to slate
- @martinalong migrated the database to a more synchronous, easier to query format

**Comment by martinalong** [comment]
## Upcoming
We're hoping to wrap up the database migration in the upcoming week or two (it's taking a while because it requires a lot of code changes as well, and we want to make sure it is bug free). Next up will be increasing user storage limits and making any adjustments needed to the analytics dashboard before launching it.

We have also begun to chip away at adding more file preview types (font files, word files, txt files, code) 

Note: The iOS app is on hold until @martinalong finishes learning UIKit because SwiftUI's functionality is too limited for a lot of things. It's basically finished, but a lot of things are not performant / hacked together in SwiftUI

## Updates 
For the week ending on Friday, Mar 19

- @akuokojnr added tagging to slates and Files
- @aminejvm added a new expanded tooltip component and fixed the bugs with the original tooltip. And worked on font files previews 
- @tarafanlin finalized designs for tagging, email verification, and font file previews
- @martinalong continued work on the database migration
- @jasonleyser added [project gutenberg import script](https://github.com/jasonleyser/project-gutenberg-importer) that will scrape the website and add files to slate

**Comment by martinalong** [comment]
## Updates
For the week ending on Friday, Mar 26

- @jasonleyser created an [importer](https://github.com/jasonleyser/project-gutenberg-importer) that will upload multiple ebooks at once to slate programatically, and migrated the [top 50 downloaded Project Gutenberg books](https://slate.host/projectgutenberg/popular-ebooks) with it
- @akuokojnr worked on tags for slates and files
- @toastts worked on API doc redesign and increasing storage to 30GB
- @tarafanlin drafting an experience map for slate
- @martinalong working on finishing up database migration

**Comment by martinalong** [comment]
## Updates

For the week ending on Friday, April 2nd

![E1316C9D-FE6C-4433-93F9-26780DA3B23D](https://user-images.githubusercontent.com/33686587/113521973-fd549300-9551-11eb-995f-bea247ce412c.jpeg)
![image](https://user-images.githubusercontent.com/33686587/113522119-faa66d80-9552-11eb-81bf-a9e8a6ac6314.png)
![image](https://user-images.githubusercontent.com/33686587/113522155-2e819300-9553-11eb-82e3-e9a661523bf9.png)
![image](https://user-images.githubusercontent.com/33686587/113522203-76a0b580-9553-11eb-8232-fe05c9b70b19.png)



- @jasonleyser and @tarafanlin shipped beta version of slate chrome extension. Test it out [here](https://github.com/slate-engineering/slate-for-chrome)!
- @akuokojnr finished adding tagging feature
- @toastts shipped API doc redesign and new codeblock component
- @aminejvm added font previews
- @tarafanlin wrapped up font previews, working on experience map for slate, and worked on calhacks hello:world event page design 
- @martinalong working on post database migration API testing

**Reply by jennijuju** [reply]
🚀🚀👏👏extension!

**Comment by martinalong** [comment]
## Updates

For the week ending on Friday, April 9th

- @martinalong is finishing up db migration code testing and moving onto making new external APIs, and adapting the old APIs to the new format
- @akuokojnr implemented kick by id (automatically logs you out of every open slate tab when you log out) and showing the cid on the data page so users can make storage deals elsewhere with their slate data
- @toastts worked on bug fixes and transferred auto-upload scripts to slate-engineering repo
- @jasonleyser added new upload options to extension (right click to upload and keyboard shortcut). And worked on functionality for updating file metadata after upload
- @tarafanlin worked on sign-in and onboarding flow designs, experience mocks for slates, and PL event page
- @harisbutt worked on ideation about the slate experience and exploring social/discovery features

**Comment by martinalong** [comment]
## Upcoming

We're in the final stretch for wrapping up the database migration, after which we can implement our new metrics dashboard and get back to working on new features full time! This coming week we'll be brainstorming which exact metrics we want to measure, and @harisbutt will be wrapping up the comprehensive redesign of our product to align with our new product direction.


## Updates

For the week ending on Friday, April 16th
- @martinalong finished the database and code migration, now rebasing changes that happened in the interim
- @tarafanlin shipped final design for new signin/signup flow and PL hackathon event page
- @aminejvm and @toastts are collaborating on implementing the new signin/signup flow as well as email and twitter authentication
- @jasonleyser finishing up polish on chrome extension

![image](https://user-images.githubusercontent.com/33686587/115127336-9344e080-9f8a-11eb-9079-db7dec335a2f.png)

**Comment by martinalong** [comment]
![image](https://user-images.githubusercontent.com/33686587/115946502-715dc900-a476-11eb-9285-6e0c941c2202.png)
![image](https://user-images.githubusercontent.com/33686587/115946506-791d6d80-a476-11eb-9801-dc944c716691.png)
![image](https://user-images.githubusercontent.com/33686587/115946504-7589e680-a476-11eb-931f-2bea23dde444.png)

## Upcoming
We've wrapped up the database migration at last and are moving onto kicking off analytics. Work on the sign-in / sign-up flow is continuing. On the design side, @tarafanlin and @harisbutt shipped the experience diagram to align with Slate's new direction. In the upcoming few months we'll be implementing new designs for each page.

## Updates
For the week ending on Friday, April 23rd
- @akuokojnr added PWA configuration for slate, and implemented optimistic uploading for audio, video, font, pdf, and images
- @tarafanlin and @harisbutt shipped the experience diagram
- @harisbutt created new chrome store assets and marketing materials for the slate extension
- @tarafanlin working on activity redesign and file and slate preview designs
- @jasonleyser shipped the extension and submitted it to the chrome store
- @martinalong shipped the database migration

**Comment by martinalong** [comment]
![image](https://user-images.githubusercontent.com/33686587/116642416-aa3ce880-a923-11eb-9eeb-076387e4f68e.png)


## Upcoming
- After the realization that there is no good place on the web to store files and links side by side, Slate is headed in the direction of "the place to save and share anything" (working description). We're prioritizing getting links into the product and exploring using Filecoin as a way to backup links for recovery in the event they are taken down
- On the analytics front: this week Martina finalized which analytics to track and what would need to be recorded in the database. She will implement tracking over the weekend and Amine will build out an analytics dashboard in metabase (and possibly observableHQ as well) next week once he's done with the twitter integration for sign-in/sign-up

## Updates
For the week ending on Friday, April, 30th

- @jasonleyser shipped the [extension](https://chrome.google.com/webstore/detail/slate/gloembacbehhbfbkcfjmloikeeaebnoc) to the Chrome store and is helping get text into Slate (think highlights, or a read-it-later like Instapaper or Pocket) with a working demo of converting any website text into a markdown file in Slate that preserves the formatting
- @martinalong implemented changes that allows users to access internal pages like explore, search, slates, and profiles without an account
- @harisbutt worked on email designs for the upcoming email verification, ideation around link objects, and font preview marketing assets
- @tarafanlin shipped the new activity and navigation bar designs, made design system updates, and shipped a PL event page
- @aminejvm shipped front end for the new sign-in/sign-up flow, and is now working on the twitter integration

**Comment by martinalong** [comment]
![image](https://user-images.githubusercontent.com/33686587/117381057-00fc7200-ae90-11eb-877f-7d52499f097e.png)

![image](https://user-images.githubusercontent.com/33686587/117380998-e75b2a80-ae8f-11eb-8a3a-49f3c9ff7db2.png)


## Upcoming
Up next we will be finalizing and implementing new activity and carousel designs, and wrapping up email/twitter verified sign-ins. All legacy accounts will be prompted to verify an email before continuing the next time they log in. In the near future will be updating the storage deal and archiving sections.

## Updates 
Updates for the week ending on Thursday, May 6th
- Amine finished the new Slate metrics dashboard. You can check it out [here](http://slate-metabase-vze2.onrender.com/public/dashboard/883d36e9-c09d-4b3b-ae17-0f7007cf8258)
- Tara finished designs for navigation, activity, and carousel view
- Jason added a new confirmation popup and updated button styles, as well as working on extension UI fixes
- Haris created assets and copy for verification and welcome emails, as well as the font support release. He also designed the link object spec
- Martina changed the way routing is done throughout the app to allow onClick based client side routing to also be used as clickable links, implemented the new navigation bar, and revamped how tab routing is done

**Comment by martinalong** [comment]
![image](https://user-images.githubusercontent.com/33686587/118208209-31f21f00-b41b-11eb-8e69-1498864569d8.png)


## Upcoming
Up next is implementing the new activity and carousel designs, as well as revamping the Filecoin section of Slate. That will be followed by adding links to Slate.

## Updates
- Jason worked on the firefox version of the Slate extension and app store submission
- Tara designed the new carousel view, profile view rough draft, and mobile activity view
- Martina shipped a new routing system for carousel and tabs which maintains state upon refresh better and loads upon first render (for better SEO), implemented the new nav bar design, and implemented a new Link component which allows us to maintain client side routing while allowing links to behave like anchor tags upon irregular clicks (right clicks, ctrl + click, etc) and for the purposes of search engine scraping
- Haris shipped the updated carousel view with Tara, and did the tagging release assets which will be going out tomorrow
- Amine is off today for a national holiday (so can't give his weekly update) and Daniel is OOO for a while to study for exams

**Comment by martinalong** [comment]
## Upcoming
Next week we'll get started implementing new activity designs now that the backend and designs are ready. On the design side, we will be brushing up how links will be presented in the full size view and solidifying the design for profiles. We will also be adding a few more analytics now that the data necessary for them is being tracked in the database.

## Updates
- Tara and Haris shipped link object preview and carousel view, and also worked on a v3 for profiles (still in progress). Haris also shipped a design for text snippets from the extension. 
- Shariq worked on more secure password transmission format, and on a strategy for moving legacy users to the new format
- Jason submitted bug fixes for extension to app store, as well as cross browser compatibility so now the extension is on safari and firefox as well! He also added text snippeting to a markdown file to the extension
- Amine shipped Twitter authentication for sign-in/sign-up
- Martina shipped better meta tags for SEO, post database migration removal of old data (now that we are sure things are working properly), database tracking necessary for activity and for analytics, adding likes, and database optimization for quicker reads of counts for followers/subscribers/likes/downloads/slates/etc, and change in the behavior for how privacy is tracked (also a speed optimization)

**Comment by martinalong** [comment]
![image](https://user-images.githubusercontent.com/33686587/119932305-d86c1300-bf37-11eb-9d1a-92c81dfd0a66.png)


## Upcoming
Amine has started work on activity and the new object formats. Design-wise, Haris and Tara are thinking through uploads, and profiles. Martina will be removing replacing reposts with save copy and then links. Shariq will be working on a thumbnail generator and other ways to speed up load time.

## Updates
Updates for the week ending on Thursday, May 27th
- Amine and Shariq shipped the new sign-in / sign-up flow. It's undergoing review and edits now
- Amine shipped lazy loading images and previewing a 3 second clip of videos
- Martina migrated the databases to accomodate faster reading of counts and added likes. Martina also shipped [the next iteration of analytics](https://analytics.slate.host/public/dashboard/883d36e9-c09d-4b3b-ae17-0f7007cf8258) which now includes growth rate numbers, per day numbers (rather than aggregate), DAU, and other comparisons to help us understand how users are using slate 
- Tara worked on file archive designs and object preview designs for specific object types
- Jason worked on isolating the CSS for the chrome extension so that it won't be affected by page specific CSS, and on making it so the extension doesn't refresh the page afterwards each use
- Haris worked on a content strategy doc and worked on designs for save and upload experience

**Comment by martinalong** [comment]
## Upcoming

We're wrapping up sign-in flow with email, Twitter soon and Amine is moving on to wrapping up new object preview styles and then the new activity design. Martina will be working on adding links once she finishes making edits to the sign in PR.

## Updates

Updates for the week ending on Friday, June 4th
- Amine shipped the sign in flow, it was reviewed, and edits were made. Just waiting final details before shipping to staging
- Tara shipped designs for profile and activity and worked on other PL projects 
- Jason worked on adding intercom to Slate as a help portal. He also shipped text to markdown for the extension and make it so the page does not reload upon closing
- Haris worked on designs for a new upload experience
- Martina removed reposts from slate and converted all existing reposts into save copies. She also added grouping functions for activity
- Shariq worked on adding a thumbnail generation server

**Comment by martinalong** [comment]
## Upcoming
Next week we'll be continuing work on links and finishing up activity.

## Updates
Updates for the week ending on Friday, June 20th
- Tara worked on the new designs for the "my collections" page
- Martina is working on links. Currently she has adapted the UI and backend to accomodate them, and is working on adding iframely to present widgets for supported links, and a backup for links whose iframes don't load
- Amine shipped the new object preview style, and started work on activity 2.0
- Jason added an intercom integration so that users can more easily ask for help, and created an uploading script for twitter and reddit to programatically upload to Slate. Check out the mechanical keyboards one [here](https://slate.host/mechanicalkeyboards/photos)

**Reply by jennijuju** [reply]
i honestly love reading your updates

**Comment by martinalong** [comment]
## Up next 
Next week we'll work on wrapping up links and activity feed!

## Updates
- Haris worked on revamping the upload experience
- Tara worked on the my collections page design
- Jason added gif support for the upload script, and added more [uploading cron jobs](https://slate-cron.onrender.com/list)
- Amine worked on the activity feed and wrapped up the new auth flow (will push to prod tomorrow)
- Martina is 70% done with adding link objects. Awaiting some designs and some blocking dependencies to finish them up

**Comment by martinalong** [comment]
*Forgot to click comment on this update till now! Sorry about that

## Upcoming
Next week Amine will continue work on activity, Martina will wrap up links, and Haris and Tara will continue working on upload and onboarding survey designs.

## Updates
For the week ending on Friday, July 2nd
- Tara worked on designs for an onboarding survey that will help understand where our users are coming from and what they use Slate for, as well as continuing work on the upload flow and ideating on the My Slate page
- Haris worked on upload designs
- Amine continued work on activity, and finished up the new object previews
- Martina added a function to upload a file directly from a URL with the help of Jim, as well as making various UI tweaks and fixing an issue with users being on the wrong authVersion and being unable to sign in
- Jason fixed some bugs with the intercom integration and scraped more collections from the web

### Featured account of the week: [@museosabiertos](https://slate.host/museosabiertos)

![image](https://user-images.githubusercontent.com/33686587/124341321-5a231200-db70-11eb-9f1c-27b75e080a19.png)
![image](https://user-images.githubusercontent.com/33686587/124341352-88a0ed00-db70-11eb-8ea7-20202d605136.png)

**Comment by martinalong** [comment]
![image](https://user-images.githubusercontent.com/33686587/126047839-4b14fb3b-21af-4b14-a02e-82df18e32255.png)


## Upcoming
We'll be working on adding cacheing, doing the final edits/optimizations to activity, and continuing work on the onboarding survey next week.


## Updates
For the week ending on July 16
- Martina fixed various bugs with coverImage display for links, added external endpoints needed for extension v2, and changing globalcarousel routing
- Tara wrapped up upload and carousel designs, and made some tweaks to activity design
- Jason added boringavatars (programmatically generated gradient avatars), and started converting the chrome extension to node.js
- Amine wrapped up activity (which is now undergoing review and edits), and started work on the user survey
- Haris worked on a new logo for slate, wrote up specs for chrome extension link support, and lightbox for links

**Comment by martinalong** [comment]
## Updates
For the week ending on July 23rd

- Haris, Tara, and Martina have been in deep discussion surrounding Slate's strategy and direction during the past week. Updates to come on the outcome soon!
- Amine shipped the onboarding survey backend 
- Jason worked on adding links and background uploads to the extension, and got the extension converted to node.js

**Comment by martinalong** [comment]
## Upcoming
We spent a couple weeks in a focused strategy session and came away with a more focused direction. Slate has decided we need to trim down some of the features that are no longer relevant to its goal, such as activity and toggling public/private for files. Generally, we want to move away from on-platform interaction and sharing, towards more of a personal utility. 

After trimming down unnecesary features, we'll start building slate into a personal "cache" where users can access all their saved files and links, as well as go back to links they've seen that they forgot to save, in the form of an improved browser history search complete with images and relationship graphing.


## Updates
For the week ending on Aug 6th
- Amine wrapped up new object preview styles and onboarding survey (onboarding survey is still under review). Amine also did the front end work of removing the peers and files section from profiles
- Martina wrapped up links and converted the privacy model to remove file level privacy settings. Martina also added an "add to slate by cid" endpoint as requested during HackFS
- Martina and Haris set a roadmap for the next few months
- Jason worked on the extension search functionality and grabbing browser history through the extension
- Tara got started on moodboarding for the new homepage and brainstorming on the new canvas implementation, as well as upload module
- Haris wrote out spec for my slate and search

**Comment by martinalong** [comment]
## Upcoming

We will be launching the new object preview styles and links tomorrow! This coming Thursday we'll be getting together for a brainstorming session to put our current direction in a concise one-pager. This week marks the start of our next sprint which will involve a much improved search experience which uses elastic search, a restructuring of slate to be a one-page filterable and searchable outlet to retrieve all of your "stuff", starting with links and files and moving into browser history in the next sprint. This sprint will also involve major updates to the slate extension and integrating it more closely with the website. You'll be able to search and retrieve from the extension, making it a two-way link into Slate rather than just a web-clipper.

## Updates

*Updates for the week ending on Aug 17th, 2021*

- Martina was out sick with covid the past week so didn't get much done personally (and hence the late update), but is back and feeling 100% now! Once getting back this week, Martina drafted up some work plans for the next sprint and reviewed the remaining PRs in preparation for the launch.
- Amine updated how we handle passing env variables to the front end, made design tweaks to object and collections previews, and removed slate layout to add the new object preview styles to slates. He also fixed a long standing bug where objects reappeared after deleting them in collections
- Jason worked on a new extension modal for uploading/saving, and did some research into how we can do history search
- Haris worked on a search spec and wires for the website landing page, and made the announcement assets for the upcoming launch
- Tara brainstormed the new search and filter architecture and UI, and made designs for collection privacy and sharing, and object favoriting and sharing, as well as upload and top navigation.

**Comment by martinalong** [comment]
## Upcoming
This week we got link saving shipped! Now you can save links and files side by side in slate. Up next Martina is working on adding elastic search to enable a much more powerful search, and Amine is converting the UI over to a search-centric single page experience. Jason will wrap up link saivng and then submit this version of the extension, before moving on to adding searching through the extension.

## Updates
_Updates for the week ending on Aug 27th, 2021_

- Amine resolved a dependencies issue that was keeping us from launching! He also updated the top navigation to the new style, and is working on filetype filtering
- Martina removed collection-level tags, prioritized this sprint's roadmaps, and got started planning for our conversion to elastic search
- Tara iterated on the filter navigation design, and worked on a new tagging and carousel design
- Jason worked on link saving through the extension, as well as downloads files and bookmarks uploading through the extension
- Haris shipped the links announcement and thought through the follower model

**Comment by martinalong** [comment]
## Upcoming

Next week we will be wrapping up the database migration, then getting started on moving to Estuary for storage deals. We will be finishing up the one pager and getting the textile buckets creation issue resolved. We are approaching the end of the current Slate extension sprint and will be doing testing to launch soon.

## Updates

*Updates for the week ending on Sept 10th, 2021*

- Amine improved uploading by adding queue-based uploading, adding the new uploading UI as well as the ability to retry uploads upon failure
- Jason worked on the extension account settings and shortcut settings pages. He also added the ability to detect the slate API key in cookies and automatically log users into the extension
- Haris worked on the one pager in preparation for an updated landing page
- Tara worked on dark mode and settings designs
- Martina worked on wrapping up the code edits for the database migration, performed the migration locally, and is now working on testing for bugs

**Comment by martinalong** [comment]
![image](https://user-images.githubusercontent.com/33686587/133869615-d4daca59-cc5e-46fc-bfb1-4c3f6994adb7.png)

## Upcoming
Next week Amine will be wrapping up the filtering sidebar and we will be testing it. Martina will add estuary then denormalize tagging and slate previews, and then start on Elastic search if there is time. Jason will be wrapping up the current extension sprint and we will ship it.

## Updates
_Updates for the week ending on Sept 17th, 2021_
- Amine shipped a new [upload flow](https://github.com/filecoin-project/slate/pull/944) with an improved queuing system to prevent race uploading duplicate files, and an upload summary to display upload status. He continued work on the filter sidebar and completed a sharing modal for files.
- Martina wrapped up the flattening migration and added endpoints for the slate extension
- Jason made UI updates to the extension and figured out how to isolate the CSS from the page so we no longer have irregularities! He is working on the onboarding flow for the extension.
- Haris worked on new brand and logo sketches, as well as extension and main application first touch screens
- Tara worked on extension onboarding and carousel mobile view, as well as the settings design

**Comment by martinalong** [comment]
## Upcoming
Next week we'll work on the one pager in preparation for the new landing page, prep work for adding elastic search, and polishing up the filtering sidebar.


## Updates
_Updates for the week ending on Sept 26th, 2021_
- Martina worked on a solution to the slow bucket creation and access issue. It's been pushed production and we'll see how it improves user signup speed
- Tara created new designs for carousel view and settings page
- Jason added shadow-dom into the extension which enables us to isolate the CSS from the page, and solves formatting issues
- Amine shipped the new filter sidebar and worked on PR suggestions for the upload interface
- Haris worked on our deliverables doc for this sprint, laid out the new landing page, and mocked up designs for a new logo

**Comment by martinalong** [comment]
## Upcoming

In the next week we are hoping to wrap up the current sprint for the chrome extension. Next up after that is the settings page revamp which Jason will be working on. Martina will continue work on elastic search and Amine will continue working on carousel and sidebar filter.

## Updates
_Updates for the week ending on Oct 15th, 2021_

- Martina continued work on queries for elastic search. The core queries are complete, just working on optimizations and integrating them with our existing search update function calls
- Amine worked on some edits to the new uploading flow as well as the new carousel design
- Tara designed a way to integrate the directory with the new website layout, as well as a new look for profiles and revision for filter bar
- Jason made more visual edits to the extension and is looking into how we can bookmark by keyboard shortcut without first initializing the chrome extension
- Haris continued work on the slate landing page and spec'd out the onboarding touchpoints with Tara
